File: C/Grace/Grace.c
#include <stdio.h>

#define FILENAME "Grace/Grace_kid.c"
#define CODE "#include <stdio.h>%c%c#define FILENAME %cGrace/Grace_kid.c%c%c#define CODE %c%s%c%c#define GRACE() int main() { FILE *f = fopen(FILENAME, %cw%c); if (f) { fprintf(f, CODE, 10, 10, 34, 34, 10, 34, CODE, 34, 10, 34, 34, 10, 10, 10, 10); fclose(f); } return 0; }%c%c// Grace comment%cGRACE()%c"
#define GRACE() int main() { FILE *f = fopen(FILENAME, "w"); if (f) { fprintf(f, CODE, 10, 10, 34, 34, 10, 34, CODE, 34, 10, 34, 34, 10, 10, 10, 10); fclose(f); } return 0; }

// Grace comment
GRACE()

File: Python/Grace/Grace.py
FILENAME = "Grace/Grace_kid.py"
CODE = "FILENAME = %cGrace/Grace_kid.py%c%cCODE = %c%s%c%c%cdef grace():%c	with open(FILENAME, %cw%c) as f:%c		f.write(CODE %% (34, 34, 10, 34, CODE, 34, 10, 10, 10, 34, 34, 10, 10, 10, 10, 10))%c%c# Grace comment%cgrace()%c"

def grace():
	with open(FILENAME, "w") as f:
		f.write(CODE % (34, 34, 10, 34, CODE, 34, 10, 10, 10, 34, 34, 10, 10, 10, 10, 10))

# Grace comment
grace()
